Source/RapyutaSimulationPlugins/Private/Drives/RobotVehicleMovementComponent.cpp

```diff
@@ -149,6 +347,72 @@ void URobotVehicleMovementComponent::TickComponent(float InDeltaTime,
 
 FTransform URobotVehicleMovementComponent::GetOdomTF() const
 {
-    return FTransform(OdomData.pose_pose_orientation,
-                      FVector(OdomData.pose_pose_position_x, OdomData.pose_pose_position_y, OdomData.pose_pose_position_z));
+    FTransform TF;
+    FVector Pos(OdomData.pose_pose_position_x, OdomData.pose_pose_position_y, OdomData.pose_pose_position_z);
+    TF.SetTranslation(Pos);
+    TF.SetRotation(OdomData.pose_pose_orientation);
+
+    return TF;
+}
+
+void URobotVehicleMovementComponent::InitMovementComponent()
+{
+    InitOdom();
+
+    TArray<UActorComponent*> ActorContactPoints = PawnOwner->GetComponentsByTag(USceneComponent::StaticClass(), "ContactPoint");
+    for (auto ACP : ActorContactPoints)
+    {
+        USceneComponent* SCP = Cast<USceneComponent>(ACP);
+        if (SCP)
+            ContactPoints.Add(SCP);

```

Comment:
Should `GetComponentsByTag(USceneComponent::StaticClass())` already assure resulting components' type as `USceneComponent` thus probably no need for the check?